### ISSUE TITLE
Patch for CVE-2024-47889 might cause a `NoMethodError`

### ISSUE THREAD

**Issue by makmic** [rails/rails#53338]
Hi :wave:,

while reviewing the [patch](https://github.com/rails/rails/commit/be898cc996986decfe238341d96b2a6573b8fd2e) for CVE-2024-47889 for Rails LTS we found that it can cause a `NoMethodError` in a specific case (see attached test below).

```rb
      splits = formatted.split(/(\*+|\#+)/)
      while line = splits.shift
        if line.start_with?("*", "#") && splits[0].start_with?(" ")
```

This code assumes that another split (`splits[0]`) has to exist after calling `splits.shift`. It should probably be adjusted like so:
```rb
        if line.start_with?("*", "#") && splits[0] && splits[0].start_with?(" ")
```

### Steps to reproduce

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "rails", "7.2.1.1"
end

require "action_mailer/railtie"
require "minitest/autorun"

class BugTest < ActionMailer::TestCase

  def helper
    Object.new.extend(ActionMailer::MailHelper)
  end

  def test_block_format
    assert_equal "  *", helper.block_format("* ")
  end

end
```

### Expected behavior

`#block_format` should not crash when given a list item without any content like `* `. The test case above should succeed.

### Actual behavior

The test crashes with a `NoMethodError`

```
  1) Error:
BugTest#test_block_format:
NoMethodError: undefined method `start_with?' for nil
    /home/michael/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/actionmailer-7.2.1.1/lib/action_mailer/mail_helper.rb:31:in `block_format'
    cve-testcase.rb:21:in `test_block_format'

1 runs, 0 assertions, 0 failures, 1 errors, 0 skips
```

### System configuration
**Rails version**: 7.2.1.1

**Ruby version**: 3.3.5

(CC @jhawthorn  - and thanks for the patch in the first place!)

**Comment by Dome-GER** [comment]
I can confirm this for Rails `6.1.7.9` and Ruby `3.3.5`: 

```
NoMethodError
undefined method `start_with?' for nil (NoMethodError)

        if line.start_with?("*", "#") && splits[0].start_with?(" ")
```

For us, this happens when parsing markdown, such as `"**bold**` but there are other scenarios as well.

**Comment by makmic** [comment]
I could prepare two PRs against `6-1-stable` and `7-2-stable` addressing this bug if it helps you closing this issue.

**Comment by jhawthorn** [comment]
This is fixed in 6.1.7.10, 7.0.8.6, 7.1.4.2, and 7.2.1.2

**Comment by jgraichen** [comment]
@jhawthorn One use case could be using `#block_format` for Markdown in the plain text part of an email. It would wrap paragraphs and indent list items.
